File: Interface Gráfica - Tela de Criação de Herói/Java/Mago.java
class Mago extends Personagem {
    protected int mana;
    private static final int CUSTO_FEITICO = 10;

    public Mago(String nome, int vida, int nivel, int mana) {
        super(nome, vida, nivel);
        setMana(mana);
    }

    // Getters
    public int getMana() {
        return (mana);
    }

    // Setters
    public void setMana(int mana) {
        if (mana >= 0) {
            this.mana = mana;
        } else {
            throw new IllegalArgumentException(
                "Mana inválida! Insira um valor maior ou igual a 0 (valor informado: " + mana + ")"
            );
        }
    }

    // Metodos
    
    @Override
    public void ficha() {
        super.ficha();
        System.out.printf("%nMana: %d%n", mana);
    }

    @Override
    public String habilidade() {
        return nome + " usa Rajada Arcana";
    }

    public String lancarFeitico() throws SemManaException {
        if (mana < CUSTO_FEITICO) {
            throw new SemManaException(mana);
        }
        mana -= CUSTO_FEITICO;
        return nome + " usa Rajada Arcana! (mana restante: " + mana + ")";
    }

}

File: Interface Gráfica - Tela de Criação de Herói/Python/mago.py
from personagem import Personagem
from sem_mana_exception import SemManaException


class Mago(Personagem):
    CUSTO_FEITICO = 10

    def __init__(self, nome, vida, nivel, mana):
        super().__init__(nome, vida, nivel)
        self.setMana(mana)

    # Getters
    def getMana(self):
        return self.mana

    # Setters
    def setMana(self, mana):
        if mana >= 0:
            self.mana = mana
        else:
            raise ValueError(
                f"Mana inválida! Insira um valor maior ou igual a 0 (valor informado: {mana})"
            )

    # Metodos
    def ficha(self):
        super().ficha()
        print(f"\nMana: {self.mana}")

    def habilidade(self):
        return f"{self.nome} usa Rajada Arcana"

    def lancarFeitico(self):
        if self.mana < self.CUSTO_FEITICO:
            raise SemManaException(self.mana)
        self.mana -= self.CUSTO_FEITICO
        return f"{self.nome} usa Rajada Arcana! (mana restante: {self.mana})"
